src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -684,7 +684,7 @@ private PreLoginHandshakeStatus ConsumePreLoginHandshake(bool encrypt, bool trus
             byte[] payload = new byte[_physicalStateObj._inBytesPacket];
 
             Debug.Assert(_physicalStateObj._syncOverAsync, "Should not attempt pends in a synchronous call");
-            result = _physicalStateObj.TryReadByteArray(payload, 0, payload.Length);
+            result = _physicalStateObj.TryReadByteArray(payload.AsSpan(0), payload.Length);
```

Comment:
You can just call the zero parameter overload `payload.AsSpan()`, no need to provide a 0 offset.

Reply:
Which is why I asked about span conventions before I submitted the changes, ah well.